lib/rules/function-url-quotes/index.mjs

```diff
@@ -54,48 +58,83 @@ const rule = (primary, secondaryOptions, context) => {
 		root.walkDecls(checkDeclParams);
 
 		/**
-		 * @param {import('postcss').Declaration} decl
+		 * @param {Declaration|AtRule} node
+		 * @param {FunctionNode} fn
+		 * @param {'add'|'remove'} mutateType
+		 * @param {ParsedValue} parsedValue
+		 */
+		function fixer(node, fn, mutateType, parsedValue) {
+			return () => {
+				switch (mutateType) {
+					case 'add':
+						addQuotes(fn);
+						break;
+					case 'remove':
+						removeQuotes(fn);
+						break;
+				}
+
+				switch (node.type) {
+					case 'decl':
+						node.value = parsedValue.toString();
+						break;
+					case 'atrule':
+						node.params = parsedValue.toString();
+						break;
+				}
+			};
+		}
+
+		/** @typedef {[Declaration|AtRule, FunctionNode, 'add'|'remove']} ArgsArray */
+		/** @typedef {Array<[Omit<Problem, 'node'>, ArgsArray]>} ReportQueue */
+
+		/**
+		 * @param {AtRule|Declaration} node
+		 * @param {string} source
+		 * @param {number} startIndex
 		 */
+		function complain(node, source, startIndex) {
+			/** @type {ReportQueue} */
+			const reportQueue = [];
+			const parsedValue = functionArgumentsSearch(source, /^url$/i, (args, index, funcNode) => {
```

Comment:
You don't have access to `parsedValue` inside the callback.

Reply:
[suggest] How about adding `parsedValue` to the callback's arguments in `functionArgumentsSearch()` like this?

```diff
--- a/lib/utils/functionArgumentsSearch.mjs
+++ b/lib/utils/functionArgumentsSearch.mjs
@@ -15,7 +15,7 @@ import { assert, isRegExp, isString } from './validateTypes.mjs';
  *
  * @param {string} source
  * @param {string | RegExp} functionName
- * @param {(expression: string, expressionIndex: number, funcNode: valueParser.FunctionNode) => void} callback
+ * @param {(expression: string, expressionIndex: number, funcNode: valueParser.FunctionNode, parsedValue: valueParser.ParsedValue) => void} callback
  * @returns {valueParser.ParsedValue}
  */
 export default function functionArgumentsSearch(source, functionName, callback) {
```

Reply:
> How about adding parsedValue to the callback's arguments in functionArgumentsSearch()

Search for it in the codebase.
i.e. 4 other rules uses it as well

Reply:
Right. I think there is no effect on other rules, even if adding an argument to the callback.

Reply:
> And what you are proposing is not possible.

How about this? 👇🏼 We can simplify the code around `reportQueue` without any effects on other rules:

```diff
--- a/lib/rules/function-url-quotes/index.mjs
+++ b/lib/rules/function-url-quotes/index.mjs
@@ -94,23 +94,17 @@ const rule = (primary, secondaryOptions) => {
 		 * @param {number} startIndex
 		 */
 		function complain(node, source, startIndex) {
-			/** @type {ReportQueue} */
-			const reportQueue = [];
-			const parsedValue = functionArgumentsSearch(source, /^url$/i, (args, index, funcNode) => {
+			functionArgumentsSearch(source, /^url$/i, (args, index, funcNode, parsedValue) => {
 				const object = checkArgs(args, startIndex + index, funcNode);
 
 				if (object) {
 					const { messageType, ...problem } = object;
 
-					reportQueue.push([problem, [node, funcNode, messageType]]);
+					const fix = fixer(node, funcNode, messageType, parsedValue);
+
+					report({ ...problem, node, fix });
 				}
 			});
-
-			reportQueue.forEach(([problem, argsArray]) => {
-				const fix = fixer.apply(null, [...argsArray, parsedValue]);
-
-				report({ ...problem, node, fix });
-			});
 		}
 
 		/** @param {Declaration} decl */
diff --git a/lib/utils/functionArgumentsSearch.mjs b/lib/utils/functionArgumentsSearch.mjs
index 044ad0522..ff49cce34 100644
--- a/lib/utils/functionArgumentsSearch.mjs
+++ b/lib/utils/functionArgumentsSearch.mjs
@@ -15,11 +15,13 @@ import { assert, isRegExp, isString } from './validateTypes.mjs';
  *
  * @param {string} source
  * @param {string | RegExp} functionName
- * @param {(expression: string, expressionIndex: number, funcNode: valueParser.FunctionNode) => void} callback
+ * @param {(expression: string, expressionIndex: number, funcNode: valueParser.FunctionNode, parsedValue: valueParser.ParsedValue) => void} callback
  * @returns {valueParser.ParsedValue}
  */
 export default function functionArgumentsSearch(source, functionName, callback) {
-	return valueParser(source).walk((node) => {
+	const parsedValue = valueParser(source);
+
+	return parsedValue.walk((node) => {
 		if (node.type !== 'function') return;
 
 		const { value } = node;
@@ -36,6 +38,6 @@ export default function functionArgumentsSearch(source, functionName, callback)
 		const parenLength = 1; // == '('
 		const expressionIndex = node.sourceIndex + value.length + parenLength;
 
-		callback(expression, expressionIndex, node);
+		callback(expression, expressionIndex, node, parsedValue);
 	});
 }
```

Reply:
This is better and will permit the other rules to have fixes added more easily.